### ISSUE TITLE
UTF-8 detected as ISO-8859-1

### ISSUE THREAD

**Issue by ferrarimarco** [editorconfig-checker/editorconfig-checker#627]
Hi, thanks for supporting the community with this tool.

I'm experiencing an issue which I believe is a false positive.

I have a file that contains ANSI escape sequences (such as terminal colors `\x1b[32m...`) and 7-bit ASCII characters.

editorconfig-checker v4.0.1 detects this file's encoding as `ISO-8859-1` instead of `utf-8`.

`editorconfig-checker` fails with:

```text
wrong character encoding ("ISO-8859-1" instead of "utf-8")
```

## Steps to Reproduce

1. Create a minimal .editorconfig:

   ```ini
    root = true

    [*]
    charset = utf-8
    ```

2. Create a file containing ANSI color escape sequences:
 
    ```
    printf '\x1b[32mHello World\x1b[0m\n' > test-ansi.txt
    ```

3. Run editorconfig-checker:
 
    ```
    editorconfig-checker test-ansi.txt
    ```

## Expected Behavior

The file should pass validation with exit code `0`.

Every byte in test-ansi.txt is strictly < 0x80 (ASCII range), which makes it valid UTF-8 per RFC 3629.

The escape character (0x1B) is a standard 7-bit ASCII/C0 control character, not an indicator of ISO-8859-1.

## Actual Behavior

editorconfig-checker reports:

```
test-ansi.txt:
    wrong character encoding ("ISO-8859-1" instead of "utf-8")

1 errors found
```

Thanks for the support!

**Comment by anandghegde** [comment]
I'd like to fix this. It looks like it came in with #598: a file that is plain ASCII apart from the ESC byte (0x1B) in ANSI color codes is now detected as ISO-8859-1. Since pure 7-bit ASCII is valid UTF-8, the fix I have in mind is to treat all-ASCII content (control characters included) as UTF-8 before trying other encodings, keeping #598's check for real Latin-1 bytes. I'd add a test with the `\x1b[32mHello World\x1b[0m` example from the issue. Would a PR be welcome, or is someone already on it?

**Comment by klaernie** [comment]
@anandghegde we would gladly welcome a PR, there is nobody already working on it.

**Comment by anandghegde** [comment]
Thanks @klaernie — opened #629.

It turned out slightly wider than my first guess. The ASCII branch in `Detect` requires *no* C0 character at all, so your file falls through to a branch that returns `ISO-8859-1` for anything without a byte `>= 0x80`. Everything reaching that branch is 7-bit, so the fix is to report ASCII there.

The one wrinkle is that ISO-2022 encodings are 7-bit too, so I kept a positive detection of those (guarded on a decoder existing for them). That incidentally fixes a second thing: the old branch was overwriting chardet's correct `ISO-2022-JP` detection at 99% confidence with `ISO-8859-1`, so four existing ISO-2022-JP fixtures are now labelled correctly.

Seven `test-results.json` expectations change as a result — all listed in the PR with reasoning, since they were recorded from the buggy behaviour. Happy to split the ISO-2022 part out if you'd rather review the ANSI-color fix on its own.

**Comment by ferrarimarco** [comment]
Thanks for fixing this issue!
